CMakeLists.txt

```diff
@@ -94,6 +94,36 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 message(STATUS "Supported C compiler features = ${CMAKE_C_COMPILE_FEATURES}")
 message(STATUS "Supported C++ compiler features = ${CMAKE_CXX_COMPILE_FEATURES}")
 
+# Enable some warnings if we know the compiler.
+if(MSVC)
+  add_compile_options(/W4 /analyze)
+  add_compile_options(/wd4100)  # unreferenced formal parameter
+  add_compile_options(/wd4267)  # narrowing conversion
+  add_compile_options(/wd4244)  # narrowing conversion
+  add_compile_options(/wd4127)  # conditional expression is constant
+  add_compile_options(/wd4995)  # #pragma deprecated
+  add_compile_options(/wd4018)  # signed/unsigned compare
+  add_compile_options(/wd4310)  # cast truncates constant value
+  add_compile_options(/wd4389)  # signed/unsigned compare
+  add_compile_options(/wd4245)  # signed/unsigned assign/return/function call
+  add_compile_options(/wd4200)  # nonstandard extension used: zero-sized array in struct/union
+  add_compile_options(/wd4702)  # unreachable code
+  add_compile_options(/wd6340)  # unsigned int passed to signed parameter
+  add_compile_options(/wd6326)  # potential comparison of a constant with another constant
+
+  # TODO(iphydf): Look into these
+  add_compile_options(/wd4996)  # use WSAAddressToStringW() instead of WSAAddressToStringA()
+  add_compile_options(/wd6255)  # don't use alloca
+  add_compile_options(/wd6385)  # reading invalid data
+  add_compile_options(/wd6001)  # using uninitialized memory
+  add_compile_options(/wd6101)  # returning uninitialized memory
+  add_compile_options(/wd6386)  # buffer overrun
+  add_compile_options(/wd6011)  # NULL dereference
+  add_compile_options(/wd6031)  # sscanf return value ignored
+  add_compile_options(/wd6387)  # passing NULL to fwrite
+  add_compile_options(/wd28159) # use GetTickCount64 instead of GetTickCount
```

Comment:
this can be removed if rebased on master

Reply:
Done.